compute_results.py

```diff
@@ -0,0 +1,219 @@
+import sys, getopt
+
+def main(argv):
+    bruxism = False
+    mema = False
+    overwrite=False
+    try:
+        opts, args = getopt.getopt(argv,"hb:m:o:",["bruxism=","mema=", "overwrite="])
+    except getopt.GetoptError:
+        print(f'compute_results.py --bruxism <boolean> --mema <boolean> --overwrite <boolean>')
+        sys.exit(2)
+    for opt, arg in opts:
+        if opt == '-h':
+            print(f'compute_results.py --bruxism <boolean> --mema <boolean>')
+            sys.exit()
+        elif opt in ("-b", "--bruxism"):
+            bruxism = arg == 'True'
+        elif opt in ("-m", "--mema"):
+            mema = arg == 'True'
+        elif opt in ("-o", "--overwrite"):
+            overwrite = arg == 'True' # re-compute the results even if files exits
+    print(f'Performs Bruxism: <{bruxism}>')
+    print(f'Performs MEMA: <{mema}>')
+    print(f'Will overwrite existing results: <{overwrite}>')
+
+    return bruxism, mema, overwrite
+
+
+
+print("config loaded.")
+
+if __name__ == "__main__":
+    import os
+    import os.path
+    import mne
+    import numpy as np
+    import pandas as pd
+    import warnings
+    from time import time
+    from tinnsleep.reports import reporting, generate_MEMA_report, generate_bruxism_report, preprocess
+    from tinnsleep.utils import crop_to_proportional_length
+    from tinnsleep.config import Config
+
+    bruxism, mema, OVERWRITE_RESULTS = main(sys.argv[1:])
+    os.chdir(os.path.join(os.path.dirname(os.path.abspath(__file__)), "notebooks/"))
+    # Setting parameters
+    EDF_list = Config.bruxisme_files
+    THR_classif_bruxism = [[0, 2], [0, 3], [0, 4], [0, 5]]
+    THR_classif_MEMA = [[0, 3], [0, 3.5], [0, 4]]
+
+    window_length_bruxism = 0.25  # in seconds (all epoch duration will be computed from here, might not be exactly this value because of rounding)
+    duration_factor_Impedance = 1  # how many time window_length
+    duration_factor_MEMA = 4  # how many time window_length
+    duration_factor_OMA = 4  # how many time window_length
+    n_adaptive_bruxism = -120  # number of seconds for adaptive (negative for forward-backward adaptive scheme)
+    n_adaptive_MEMA = -60  # number of seconds for adaptive (negative for forward-backward adaptive scheme)
+    delim = 3  # maximal time interval between bursts to merge episode in seconds
+    results_file_MEMA =  "data/reports_and_datas_MEMA.pk"
+    results_file_bruxism = "data/reports_and_datas_bruxism.pk"
+
+
+    # Dictionnary of known names of the Airflow
+    mapping = {"Airflow": "MEMA"}
+    print("parameters set")
+
+    # Importing personnalized parameters for dataset
+    mema_files = pd.read_csv("data/mema_files.csv", engine='python', sep="; ")["files_with_mema"].values
+    dico_chans = pd.read_pickle("data/valid_chans_THR_imp.pk").to_dict("list")
```

Comment:
We should maybe verify one last time that the channel selection is accurately done

Reply:
added TODO